### PR TITLE
Create stepper

### PR DESCRIPTION
**Pull request by kabiroberai**
## Notion ticket link
[Create intake form stepper component](https://www.notion.so/uwblueprintexecs/Create-intake-form-stepper-component-90e87b8bad9e48a684bbf3c985e2291c)

## Implementation description
* `Stepper` is a React component, which takes a list of pages (just strings for now) + an active page index
* The stepper is laid out using the CSS grid API, though this can be replaced with flexbox if compatibility with super old browsers is a concern; let me know if so.

## Steps to test
1. Navigate to <frontend>/intake
2. Go back and forth between pages, resize, etc

## What should reviewers focus on?
* The margins and padding currently use hardcoded pixel values, since that's what most other components appear to do as well. This is just food for thought atm, but we might want to switch to em/rem eventually for better responsiveness.
* The UI breaks on mobile devices but this seems to be okay per the spec. If this requirement has changed, I can update the PR to improve the mobile layout.

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

https://user-images.githubusercontent.com/12706786/190971336-0823d048-dcd8-4f46-ba71-087f67e3e877.mov

### CODE REVIEW

File: frontend/src/components/pages/IntakePage.tsx [hunk 1]
```diff
@@ -44,57 +45,52 @@ const Intake = (): React.ReactElement => {
...
+        primaryTitle="Initiate New Case"
+        secondaryTitle="Case Management"
+      />
+      <Box textAlign="center" padding="30px 0 40px 0">
+        <Stepper
+          pages={[
+            "Case referral",
+            "Court information",
+            "Individual details",
+            "Program details",
```
Comment (kabiroberai) [inline frontend/src/components/pages/IntakePage.tsx:86]: I don't think we have the program details page yet fwiw – in the UI this simply materializes as there being no way to navigate to that page; the stepper handles it just fine.

**Comment by kabiroberai** [comment]
updated the reviewers since it seems Atharva jumped ship, pun entirely intended

File: frontend/src/components/intake/Stepper.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                  />
+                  <GridItem gridRowStart={2} />
+                </>
+              )}
+              <GridItem
+                gridRowStart={1}
+                justifySelf="center"
+                sx={{ "svg path": { stroke: color } }}
+              >
+                {isComplete ? (
```
Comment (jennchenn) [inline frontend/src/components/intake/Stepper.tsx:48]: for the check icon, do you mind using [this](https://feathericons.com/?query=check-circle) instead

File: frontend/src/components/pages/IntakePage.tsx [hunk 3]
```diff
@@ -44,57 +45,52 @@ const Intake = (): React.ReactElement => {
...
+          ]}
+          activePage={step - 1}
```
Comment (jennchenn) [inline frontend/src/components/pages/IntakePage.tsx:88]: do you think it makes sense to keep the stepper 0-indexed despite having to do this conversion here to get the `activePage`?

**Review by jennchenn — COMMENTED** [review]
looking good!!

File: frontend/src/components/pages/IntakePage.tsx (hunk 3 above)
Comment (kabiroberai) [reply frontend/src/components/pages/IntakePage.tsx:88]: I don't think the `Stepper` component should be dependent on the implementation details of a client; in fact I'd suggest making `step` zero-indexed too but that's a separate discussion.

File: frontend/src/components/intake/Stepper.tsx (hunk 2 above)
Comment (kabiroberai) [reply frontend/src/components/intake/Stepper.tsx]: neat! done.

**Review by jennchenn — APPROVED** [review]
looks good to me! nice work

**Pull request merged**